src/services/stats/stats-router.test.ts

```diff
@@ -0,0 +1,447 @@
+import { beforeEach, describe, expect, it } from "@jest/globals";
+import { StatusCodes } from "http-status-codes";
+import { Database } from "../../database";
+import { Role } from "../auth/auth-models";
+import { getAsStaff, get } from "../../../testing/testingTools";
+import { getCurrentDay } from "../checkin/checkin-utils";
+
+describe("GET /stats/check-in", () => {
+    beforeEach(async () => {
+        await Database.ATTENDEE.deleteMany({});
+    });
+
+    it("should return correct count for checked-in attendees", async () => {
+        await Database.ATTENDEE.create([
+            {
+                userId: "a1",
+                name: "Ritam",
+                email: "ritam@test.com",
+                hasCheckedIn: true,
+            },
+            {
+                userId: "a2",
+                name: "Nathan",
+                email: "nathan@test.com",
+                hasCheckedIn: true,
+            },
+            {
+                userId: "a3",
+                name: "Timothy",
+                email: "timothy@test.com",
+                hasCheckedIn: false,
+            },
+        ]);
+
+        const response = await getAsStaff("/stats/check-in").expect(
+            StatusCodes.OK
+        );
+
+        expect(response.body).toEqual({ count: 2 });
+    });
+
+    it("should return 0 if no attendees are checked in", async () => {
+        await Database.ATTENDEE.create([
+            {
+                userId: "a1",
+                name: "Ritam",
+                email: "ritam@test.com",
+                hasCheckedIn: false,
+            },
+        ]);
+
+        const response = await getAsStaff("/stats/check-in").expect(
+            StatusCodes.OK
+        );
+
+        expect(response.body).toEqual({ count: 0 });
+    });
+
+    it("should return 401 for unauthenticated users", async () => {
+        await get("/stats/check-in").expect(StatusCodes.UNAUTHORIZED);
+    });
+
+    it("should return 403 for users without STAFF access", async () => {
+        await get("/stats/check-in", Role.enum.USER).expect(
+            StatusCodes.FORBIDDEN
+        );
+    });
+});
+
+describe("GET /stats/merch-item/:PRICE", () => {
+    beforeEach(async () => {
+        await Database.ATTENDEE.deleteMany({});
+    });
+
+    it("should return correct count for people with points threshold", async () => {
+        await Database.ATTENDEE.create([
+            {
+                userId: "a1",
+                name: "Ritam",
+                email: "ritam@test.com",
+                points: 10,
+            },
+            {
+                userId: "a2",
+                name: "Nathan",
+                email: "nathan@test.com",
+                points: 25,
+            },
+            {
+                userId: "a3",
+                name: "Timothy",
+                email: "timothy@test.com",
+                points: 20,
+            },
+        ]);
```

Comment:
Same point as before - extract this out. Don't duplicate things when you don't need to, that way tests are easier to modify in the future.